packages/ui/src/services/breakpoints/plugin/create-service.ts

```diff
@@ -0,0 +1,138 @@
+import { App, ref, Ref, computed, watch } from 'vue'
+
+import { warn } from '../../utils'
+import { getGlobalProperty } from '../../../vuestic-plugin/utils'
+import { addOrUpdateStyleElement } from '../../dom-functions'
+
+import { GlobalConfig } from '../../global-config/types'
+import { ThresholdsKeys, BreakpointsConfig, BodyClass } from '../types'
+
+export const createBreakpointsConfigPlugin = (app: App) => {
+  const globalConfig: Ref<GlobalConfig> | undefined = getGlobalProperty(app, '$vaConfig')?.globalConfig
+  if (!globalConfig) {
+    warn('createBreakpointsConfigPlugin: globalConfig is not defined!')
+    return {}
+  }
+
+  const breakpointsConfig: BreakpointsConfig | undefined = globalConfig.value.breakpoints
+  if (!breakpointsConfig) {
+    warn('createBreakpointsConfigPlugin: breakpointsConfig is not defined!')
+    return {}
+  }
+
+  if (!breakpointsConfig.enabled) {
+    return {}
+  }
+
+  if (!breakpointsConfig.thresholds || !Object.values(breakpointsConfig.thresholds).length) {
+    warn('createBreakpointsConfigPlugin: there are no defined thresholds!')
+    return {}
+  }
+
+  const isMounted = computed(() => !!document.documentElement)
+  const windowWidth = ref<number | undefined>()
+  const windowHeight = ref<number | undefined>()
+
+  const setCurrentDocumentSizes = () => {
+    windowWidth.value = document.documentElement.clientWidth
```

Comment:
const windowSizes = reactive(...)
const window= useWindow()
watchEffect(() => {
  windowSizes.width = window.innerWidth
})

Reply:
Window attributes can't be reactive, only event listener.